github-extension/src/node/github-service-impl.ts

```diff
@@ -11,107 +11,116 @@
 
 import { injectable, inject } from "inversify";
 import { GithubService, SshKeyServer } from '../common/github-service';
-import { Repository, Credentials, User, PullRequest, Organization, Collaborator } from '../common/github-model';
+import {
+    Repository,
+    User,
+    PullRequest,
+    Organization,
+    Collaborator,
+    Properties
+} from '../common/github-model';
 
-const octokit = require('@octokit/rest')();
+const octokit = require('@octokit/rest');
 
 @injectable()
 export class GithubServiceImpl implements GithubService {
 
     constructor(@inject(SshKeyServer) protected readonly sshKeyServer: SshKeyServer) { }
 
-    async getRepository(credentials: Credentials, owner: string, repository: string): Promise<Repository> {
-        const response = await this.getConnection(credentials).repos.get({ owner: owner, repo: repository });
+    async getRepository(owner: string, repository: string, properties?: Properties): Promise<Repository> {
+        const response = await GithubServiceImpl.getConnection(properties).repos.get({ owner: owner, repo: repository });
```

Comment:
why you call GithubServiceImpl directly, looks strange to me?

Reply:
It's a specific writing for calling static methods, we can't simply call static method my the method name. In that case we have to refer to the method name through class.

Reply:
You can move the `getConnection` method to some `GitHubConnectionFactory` class.
Or make the `getConnection` as a function ([example](https://github.com/theia-ide/theia/blob/master/packages/git/src/node/test/binding-helper.ts#L26)) in a separate module and use it by its name, i.e.
`import { getConnection } from '../common/github-connection-factory'`

Reply:
Thanks, @azatsarynnyy, I'll try the variant with exported function